src/hotspot/os/linux/waitBarrier_linux.cpp

```diff
@@ -51,9 +51,9 @@ void LinuxWaitBarrier::arm(int barrier_tag) {
 void LinuxWaitBarrier::disarm() {
   assert(_futex_barrier != 0, "Should be armed/non-zero.");
   _futex_barrier = 0;
-  int s = futex(&_futex_barrier,
-                FUTEX_WAKE_PRIVATE,
-                INT_MAX /* wake a max of this many threads */);
+  intptr_t s = futex(&_futex_barrier,
```

Comment:
futex returns an int

Reply:
src/hotspot/os/linux/waitBarrier_linux.cpp:41:17: warning: conversion from 'long int' to 'int' may change value [-Wconversion]
   41 |   return syscall(SYS_futex, addr, futex_op, op_arg, nullptr, nullptr, 0);
      |          ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Reply:
I was again referring to:
```
SYNOPSIS
       #include <linux/futex.h>
       #include <sys/time.h>

       int futex(int *uaddr, int op, int val, const struct timespec *timeout,
                 int *uaddr2, int val3);
```
But why `intptr_t` here, but `int64_t` in the earlier syscall case?